.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,62 @@
+name: PR Checks
+
+permissions:
+  contents: write
+  pull-requests: write
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, reopened, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  codeql:
+    name: Run CodeQL Scan
+    runs-on: ubuntu-latest
+    concurrency: ${{ github.workflow }}-${{ matrix.language }}-${{ github.ref }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+            - javascript
+    steps:
+      - name: Run Code Scanning
+        uses: department-of-veterans-affairs/codeql-tools/codeql-analysis@main
+        with:
+          language: ${{ matrix.language }}
+
+  security:
+    name: NPM Audit
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Run NPM audit
+        run: npm audit --production --audit-level high
+
+  auto-merge:
+    name: Auto-Merge Dependabot PRs
+    needs: [codeql, security]
+    runs-on: ubuntu-latest
+    if: success() && github.event.pull_request.user.login == 'dependabot[bot]'
```

Comment:
I didn't find explicit confirmation in the github docs that `success()` will check previous jobs (as opposed to previous steps), but [this doc](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs) implies that conditional expressions like `always()` (and, it seems, `success()`), depend on the status of the jobs listed in their `needs` configuration.

Reply:
Sounds good enough to me. 👍